pkg/processor/native/utils_test.go

```diff
@@ -99,42 +100,64 @@ func TestGetStartingPointForCrop(t *testing.T) {
 	assert.Equal(t, 0, y)
 }
 
-func Test_isOpaqueWithoutOpaqueMethod(t *testing.T) {
-	im := &mockImage{opaque: false}
-	val := isOpaque(im)
-	assert.False(t, val)
-
-	im = &mockImage{opaque: true}
-	val = isOpaque(im)
+func Test_isOpaqueWithoutOpaqueMethodShouldReturnTrue(t *testing.T) {
+	img := NewMockImage(image.Rect(0, 0, 640, 480))
+	draw.Draw(img, img.Bounds(), image.Opaque, image.ZP, draw.Src)
+	val := isOpaque(img)
 	assert.True(t, val)
 }
 
+func Test_isOpaqueWithoutOpaqueMethodShouldReturnFalse(t *testing.T) {
+	w, h := 640, 480
+	img := NewMockImage(image.Rect(0, 0, w, h))
+	draw.Draw(img, img.Bounds(), image.Opaque, image.ZP, draw.Src)
+
+	cases := []struct {
+		x, y int
+	}{
+		{x: 0, y: 0},
+		{x: w / 2, y: h / 2},
+		{x: w - 1, y: h - 1},
+	}
+	for _, c := range cases {
+		// Flip only 1 bit to be transparent for each test case
+		x, y := c.x, c.y
+		img.Set(x, y, image.Transparent.C)
+		val := isOpaque(img)
+		assert.False(t, val)
+		img.Set(x, y, image.Opaque.C)
```

Comment:
Shouldn't we assert True after this?

Reply:
Yes that would be correct but we are not testing that in this block of code since it's already tested above, wdyt?

Reply:
Okay, no problem. Should be fine this way.